### PR TITLE
Add missing code, and add flexible PostgreSQL

### LINKED ISSUE #428: Make Azure Postgres Flexible Server specific deployment article.
At the end of this article:

https://learn.microsoft.com/en-us/dotnet/aspire/database/postgresql-component?tabs=dotnet-cli

We should link on to a new article which describes how to deploy a .NET Aspire application with a Postgres database resource to Azure. We have a new API for this that generates Bicep that is used by AZD.

```csharp
builder.AddPostgres("pgsql").PublishAsAzurePostgresFlexibleServer().AddDatabase(db);
```

---
[Associated WorkItem - 238748](https://dev.azure.com/msft-skilling/Content/_workitems/edit/238748)

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Ok, I got a little carried away, but these were missing.

Fixes #428

---

#### Internal previews

<details><summary><strong>Toggle expand/collapse</strong></summary><br/>

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/azureai/azureai-openai-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/azureai/azureai-openai-component.md) | [.NET Aspire Azure AI OpenAI component](https://review.learn.microsoft.com/en-us/dotnet/aspire/azureai/azureai-openai-component?branch=pr-en-us-1046) |
| [docs/azureai/azureai-search-document-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/azureai/azureai-search-document-component.md) | [.NET Aspire Azure AI Search Documents component](https://review.learn.microsoft.com/en-us/dotnet/aspire/azureai/azureai-search-document-component?branch=pr-en-us-1046) |
| [docs/caching/stackexchange-redis-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/caching/stackexchange-redis-component.md) | [.NET Aspire StackExchange Redis component](https://review.learn.microsoft.com/en-us/dotnet/aspire/caching/stackexchange-redis-component?branch=pr-en-us-1046) |
| [docs/caching/stackexchange-redis-distributed-caching-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/caching/stackexchange-redis-distributed-caching-component.md) | [docs/caching/stackexchange-redis-distributed-caching-component](https://review.learn.microsoft.com/en-us/dotnet/aspire/caching/stackexchange-redis-distributed-caching-component?branch=pr-en-us-1046) |
| [docs/caching/stackexchange-redis-output-caching-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/caching/stackexchange-redis-output-caching-component.md) | [.NET Aspire StackExchange Redis output caching component](https://review.learn.microsoft.com/en-us/dotnet/aspire/caching/stackexchange-redis-output-caching-component?branch=pr-en-us-1046) |
| [docs/database/azure-cosmos-db-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/database/azure-cosmos-db-component.md) | [.NET Aspire Azure Cosmos DB component](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/azure-cosmos-db-component?branch=pr-en-us-1046) |
| [docs/database/azure-cosmos-db-entity-framework-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/database/azure-cosmos-db-entity-framework-component.md) | [.NET Aspire Microsoft Entity Framework Core Cosmos DB component](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/azure-cosmos-db-entity-framework-component?branch=pr-en-us-1046) |
| [docs/database/mongodb-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/database/mongodb-component.md) | [docs/database/mongodb-component](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/mongodb-component?branch=pr-en-us-1046) |
| [docs/database/mysql-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/database/mysql-component.md) | [.NET Aspire MySQL database component](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/mysql-component?branch=pr-en-us-1046) |
| [docs/database/mysql-entity-framework-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/database/mysql-entity-framework-component.md) | [MySQL Entity Framework Component](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/mysql-entity-framework-component?branch=pr-en-us-1046) |
| [docs/database/oracle-entity-framework-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/database/oracle-entity-framework-component.md) | [.NET Aspire Oracle Entity Framework Component](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/oracle-entity-framework-component?branch=pr-en-us-1046) |
| [docs/database/postgresql-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/database/postgresql-component.md) | [docs/database/postgresql-component](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/postgresql-component?branch=pr-en-us-1046) |
| [docs/database/postgresql-entity-framework-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/database/postgresql-entity-framework-component.md) | [.NET Aspire PostgreSQL Entity Framework Core component](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/postgresql-entity-framework-component?branch=pr-en-us-1046) |
| [docs/database/qdrant-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/database/qdrant-component.md) | [.NET Aspire Qdrant component](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/qdrant-component?branch=pr-en-us-1046) |
| [docs/database/sql-server-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/database/sql-server-component.md) | [.NET Aspire SQL Server component](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/sql-server-component?branch=pr-en-us-1046) |
| [docs/database/sql-server-entity-framework-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/database/sql-server-entity-framework-component.md) | [.NET Aspire SqlServer Entity Framework Core component](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/sql-server-entity-framework-component?branch=pr-en-us-1046) |
| [docs/fundamentals/app-host-overview.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/fundamentals/app-host-overview.md) | [docs/fundamentals/app-host-overview](https://review.learn.microsoft.com/en-us/dotnet/aspire/fundamentals/app-host-overview?branch=pr-en-us-1046) |
| [docs/logging/seq-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/logging/seq-component.md) | [.NET Aspire Seq component](https://review.learn.microsoft.com/en-us/dotnet/aspire/logging/seq-component?branch=pr-en-us-1046) |
| [docs/messaging/azure-event-hubs-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/messaging/azure-event-hubs-component.md) | [.NET Aspire Azure Event Hubs component](https://review.learn.microsoft.com/en-us/dotnet/aspire/messaging/azure-event-hubs-component?branch=pr-en-us-1046) |
| [docs/messaging/azure-service-bus-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/messaging/azure-service-bus-component.md) | [.NET Aspire Azure Service Bus component](https://review.learn.microsoft.com/en-us/dotnet/aspire/messaging/azure-service-bus-component?branch=pr-en-us-1046) |
| [docs/messaging/kafka-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/messaging/kafka-component.md) | [docs/messaging/kafka-component](https://review.learn.microsoft.com/en-us/dotnet/aspire/messaging/kafka-component?branch=pr-en-us-1046) |
| [docs/messaging/nats-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/messaging/nats-component.md) | [.NET Aspire NATS component](https://review.learn.microsoft.com/en-us/dotnet/aspire/messaging/nats-component?branch=pr-en-us-1046) |
| [docs/messaging/rabbitmq-client-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/messaging/rabbitmq-client-component.md) | [.NET Aspire RabbitMQ component](https://review.learn.microsoft.com/en-us/dotnet/aspire/messaging/rabbitmq-client-component?branch=pr-en-us-1046) |
| [docs/security/azure-security-key-vault-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/security/azure-security-key-vault-component.md) | [.NET Aspire Azure Key Vault component](https://review.learn.microsoft.com/en-us/dotnet/aspire/security/azure-security-key-vault-component?branch=pr-en-us-1046) |
| [docs/storage/azure-storage-blobs-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/storage/azure-storage-blobs-component.md) | [.NET Aspire Azure Blob Storage component](https://review.learn.microsoft.com/en-us/dotnet/aspire/storage/azure-storage-blobs-component?branch=pr-en-us-1046) |
| [docs/storage/azure-storage-queues-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/storage/azure-storage-queues-component.md) | [docs/storage/azure-storage-queues-component](https://review.learn.microsoft.com/en-us/dotnet/aspire/storage/azure-storage-queues-component?branch=pr-en-us-1046) |
| [docs/storage/azure-storage-tables-component.md](https://github.com/dotnet/docs-aspire/blob/ae2de99bae62d0e6eaa032cd97b8589951adc594/docs/storage/azure-storage-tables-component.md) | [.NET Aspire Azure Data Tables component](https://review.learn.microsoft.com/en-us/dotnet/aspire/storage/azure-storage-tables-component?branch=pr-en-us-1046) |

</details>

### CODE REVIEW

File: docs/database/includes/postgresql-flexible-server.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+### Azure app host usage
+
+If you'd like to deploy your PostgreSQL resources to Azure, you need to install the appropriate .NET Aspire hosting package:
```
Comment (gewarren) [inline docs/database/includes/postgresql-flexible-server.md:3]:
```suggestion
To deploy your PostgreSQL resources to Azure, you need to install the appropriate .NET Aspire hosting package:
```

File: docs/fundamentals/app-host-overview.md [hunk 2]
```diff
@@ -256,6 +256,33 @@ Beyond the base resource types, <xref:Aspire.Hosting.ApplicationModel.ProjectRes
 
 For more information, see [GitHub: Aspire.Hosting.AWS library](https://github.com/dotnet/aspire/tree/main/src/Aspire.Hosting.AWS).
 
+## Execution context
+
+The <xref:Aspire.Hosting.IDistributedApplicationBuilder> exposes an execution context (<xref:Aspire.Hosting.DistributedApplicationExecutionContext>), which provides information about the current execution of the app host. This can be used to evaluate whether or not the app host is executing as run more, or as part of a publish operation. Consider the following:
```
Comment (gewarren) [inline docs/fundamentals/app-host-overview.md:261]:
```suggestion
The <xref:Aspire.Hosting.IDistributedApplicationBuilder> exposes an execution context (<xref:Aspire.Hosting.DistributedApplicationExecutionContext>), which provides information about the current execution of the app host. This context can be used to evaluate whether or not the app host is executing as "run" mode, or as part of a publish operation. Consider the following:
```

File: docs/fundamentals/app-host-overview.md [hunk 3]
```diff
@@ -256,6 +256,33 @@ Beyond the base resource types, <xref:Aspire.Hosting.ApplicationModel.ProjectRes
...
+
+- <xref:Aspire.Hosting.DistributedApplicationExecutionContext.IsRunMode%2A>: Returns `true` if the current operation is running.
+- <xref:Aspire.Hosting.DistributedApplicationExecutionContext.IsPublishMode%2A>: Returns `true` if the current operation is publishing.
+
+This can be useful when you want to conditionally execute code based on the current operation. Consider the following example that demonstrates using the `IsRunMode` property. In this case, an extension method is used to generate a stable node name for RabbitMQ for local development runs.
```
Comment (gewarren) [inline docs/fundamentals/app-host-overview.md:266]:
```suggestion
This information can be useful when you want to conditionally execute code based on the current operation. Consider the following example that demonstrates using the `IsRunMode` property. In this case, an extension method is used to generate a stable node name for RabbitMQ for local development runs.
```

File: docs/messaging/nats-component.md [hunk 4]
```diff
@@ -54,6 +52,42 @@ public class ExampleService(INatsConnection client)
...
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var nats = builder.AddNats("nats");
+
+var myService = builder.AddProject<Projects.MyService>()
+                       .WithReference(nats);
+```
+
+The `WithReference` method configures a connection in the `MyService` project named `nats`. In the _:::no-loc text="Program.cs":::_ file of `MyService`, the NATS connection can be consumed using:
```
Comment (gewarren) [inline docs/messaging/nats-component.md:85]: You could set Program.cs as file-level, folder-level, or global no-loc metadata.

**Review by gewarren — APPROVED** [review]

File: docs/messaging/nats-component.md (hunk 4 above)
Comment (IEvangelist) [reply docs/messaging/nats-component.md:85]: Yes, that's a good idea. Thank you!

File: docs/database/includes/postgresql-flexible-server.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+### [PackageReference](#tab/package-reference)
+
+```xml
+<PackageReference Include="Aspire.Hosting.Azure.PostgreSQL"
+                  Version="[SelectVersion]" />
+```
+
+---
+
+After you've installed this package, you specify that your PostgreSQL resources will be hosted in Azure by calling the <xref:Aspire.Hosting.AzurePostgresExtensions.PublishAsAzurePostgresFlexibleServer*%2A> extension method in your app host project:
```
Comment (IEvangelist) [inline docs/database/includes/postgresql-flexible-server.md:20]:
```suggestion
After you've installed this package, you specify that your PostgreSQL resources will be hosted in Azure by calling the <xref:Aspire.Hosting.AzurePostgresExtensions.PublishAsAzurePostgresFlexibleServer%2A> extension method in your app host project:
```
```suggestion
After you've installed this package, you specify that your PostgreSQL resources will be hosted in Azure by calling the <xref:Aspire.Hosting.AzurePostgresExtensions.PublishAsAzurePostgresFlexibleServer*%2A> extension method in your app host project:
```

**Pull request merged**